### PR TITLE
New query stake-snapshot integration test

### PR DESCRIPTION
**Pull request by newhoggy**
For https://github.com/input-output-hk/cardano-node/issues/4570

### CODE REVIEW

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+{- HLINT ignore "Redundant id" -}
+{- HLINT ignore "Redundant return" -}
+{- HLINT ignore "Use head" -}
+{- HLINT ignore "Use let" -}
+
+module Test.Cli.Babbage.StakeSnapshot
+  ( hprop_stakeSnapshot
+  ) where
+
+import           Cardano.CLI.Shelley.Output (QueryTipLocalStateOutput (..))
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs:19]: These imports don't adhere to the typical style found in the repo.

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  result <- execCli' execConfig
+    [ "query", "stake-snapshot"
+    , "--testnet-magic", show @Int testnetMagic
+    , "--all-stake-pools"
+    ]
+
+  let jsonResult = J.eitherDecode @J.Value (LBS.fromStrict (T.encodeUtf8 (T.pack result)))
+
+  -- Check that "pools" has three entries
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs:111]: and therefore there are three registered stake pools?

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
Minor changes

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 3]
```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE BlockArguments #-}
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs:1]: `BlockArguments` not necessary here

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs]: Reordered the imports.

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs (hunk 3 above)
Comment (newhoggy) [reply cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs]: Removed

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+module Test.Cli.Babbage.StakeSnapshot
+  ( hprop_stakeSnapshot
+  ) where
+
+import           Cardano.CLI.Shelley.Output (QueryTipLocalStateOutput (..))
+import           Cardano.Testnet
+
+import           Control.Monad (void)
+import qualified Data.Aeson as J
```
[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs]: Renamed.

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+import qualified Data.Aeson.KeyMap as KM
+import qualified Data.ByteString.Lazy as LBS
+import           Data.Monoid (Last (..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Time.Clock as DTC
+import           GHC.Stack (callStack)
+import           Prelude
```
[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs]: Moved.

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+import           Cardano.CLI.Shelley.Output (QueryTipLocalStateOutput (..))
+import           Cardano.Testnet
+
+import           Control.Monad (void)
+import qualified Data.Aeson as J
+import qualified Data.Aeson.KeyMap as KM
+import qualified Data.ByteString.Lazy as LBS
+import           Data.Monoid (Last (..))
+import qualified Data.Text as T
```
[the remark this answers is not in the document]
Comment (newhoggy) [reply cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs]: Renamed.

**Comment by jbgi** [comment]
bors try

File: cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+{- HLINT ignore "Redundant id" -}
+{- HLINT ignore "Redundant return" -}
+{- HLINT ignore "Use head" -}
+{- HLINT ignore "Use let" -}
+
+module Test.Cli.Babbage.StakeSnapshot
+  ( hprop_stakeSnapshot
+  ) where
+
+import           Prelude
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/Babbage/StakeSnapshot.hs:18]: Perfect, thanks :+1:

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**